File: core/src/org/rexellentgames/dungeon/util/file/FileReader.java
package org.rexellentgames.dungeon.util.file;

import java.io.*;

public class FileReader {
	private DataInputStream stream;

	public FileReader(String path) throws FileNotFoundException {
		this.stream = new DataInputStream(new BufferedInputStream(new FileInputStream(path), 32768));
	}

	public byte readByte() throws IOException {
		return this.stream.readByte();
	}

	public boolean readBoolean() throws IOException {
		return this.stream.readBoolean();
	}

	public short readInt16() throws IOException {
		return this.stream.readShort();
	}

	public int readInt32() throws IOException {
		return this.stream.readInt();
	}

	public String readString() throws IOException {
		byte length = this.stream.readByte();

		if (length == 0) {
			return null;
		}

		StringBuilder result = new StringBuilder();

		for (int i = 0; i < length; i++) {
			result.append(this.stream.readChar());
		}

		return result.toString();
	}

	public double readDouble() throws IOException {
		return this.stream.readDouble();
	}

	public void close() throws IOException {
		this.stream.close();
	}
}

File: BurningKnight/Util/Files/FileReader.cs
using System.IO;
using System.Text;

namespace BurningKnight.Util.Files
{
	public class FileReader
	{
		private FileStream stream;
		
		public FileReader(string path)
		{
			stream = File.OpenRead(path);
		}

		public void Close()
		{
			stream.Close();
		}

		public byte ReadByte()
		{
			return (byte) stream.ReadByte();
		}

		public short ReadShort()
		{
			// Not sure
			return (short) (ReadByte() + ReadByte() >> 4);
		}

		public int ReadInt()
		{
			return ReadShort() + ReadShort() >> 8;
		}

		public long ReadLong()
		{
			return ReadInt() + ReadInt() >> 16;			
		}

		public string ReadString()
		{
			byte len = ReadByte();

			if (len == 0)
			{
				return null;
			}

			StringBuilder builder = new StringBuilder();

			for (int i = 0; i < len; i++)
			{
				builder.Append((char) ReadByte());
			}
			
			return builder.ToString();
		}
	}
}